### DISCUSSION TITLE
How to change 'solution' title and prefix?

### DISCUSSION THREAD

**Discussion by jandermoreira** [quarto-dev/quarto-cli#10020] [Q&A]
### Description

I want to use `#exr-...`  `div`s to present problems to students, as well as `#sol-...` to provide answers.

The text is in Portuguese and here is a MWE:

```qmd
---
title: Exemplo
author: Jander
lang: pt-BR

crossref:
    exr-title: "Problema"
    exr-prefix: "Problema"
    # sol-title: "Resposta"
    # sol-prefix: "Resposta"
---

# Problemas

::: {#exr-1}
Um problema qualquer.

Veja @sol-1.
:::

# Respostas

::: {#sol-1}
Uma resposta.

Veja @exr-1.
:::
```

A first issue is that `#sol-...` labels create "Solução 1", but `@sol-...` remains "Solution" (in English).

I actually want to change both title and prefix for _solutions_ to "Resposta" (a better word in Portuguese for my case).

I can use `exr-title` and `exr-prefix` to change _exercices_ to _problems_ (problemas), but there's no `sol-title` nor `sol-prefix` to adjust solutions.

```
ERROR: Validation of YAML front matter failed.
ERROR: In file exemplo.qmd
(line 9, columns 5--14) property name sol-title is invalid
 8:     exr-prefix: "Problema"
 9:     sol-title: "Resposta"
       ~~~~~~~~~~
10:     sol-prefix: "Resposta"
ℹ The error happened in location crossref:sol-title.
ℹ Did you mean lol-title?

ERROR: In file exemplo.qmd
(line 10, columns 5--15) property name sol-prefix is invalid
 9:     sol-title: "Resposta"
10:     sol-prefix: "Resposta"
       ~~~~~~~~~~~
11: ---
ℹ The error happened in location crossref:sol-prefix.
ℹ Did you mean tbl-prefix, sec-prefix, or cor-prefix?
```

Quarto's [crossref options page](https://quarto.org/docs/reference/metadata/crossref.html) shows the available translation options, but _solutions_ aren't there. (Curiously, _proof_ is not there either...)

Is there a way to correct this or should I consider writing a filter?

**Quarto**
```
$ quarto --version
1.4.542
```

**Comment by mcanouil** [comment]
Indeed, there's no equivalent and not only for "solution".
The pattern is not the same for "solution" (and "remark"), see <https://quarto.org/docs/authoring/language.html#custom-translations> and <https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/language/_language.yml#L104-L106>.

````qmd
---
title: Exemplo
author: Jander
lang: pt-BR
language:
  crossref-exr-title: "Problemaaaa"
  environment-solution-title: "Resposta"
---

# Problemas

::: {#exr-1}
Um problema qualquer.

Veja @sol-1.
:::

# Respostas

::: {#sol-1}
Uma resposta.

Veja @exr-1.
:::
````

cc @cscheid From what I've seen, solution/remark (and proof) were special environments, but the documentation currently implies that they are of the same kind as the others: https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs
`proof` is still different. But not "solution" and "remark".
Should "solution" and "remark" work as "exercise", "theorem", etc. or should they be more explicitly separated from the others in the documentation (https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs and https://quarto.org/docs/reference/metadata/crossref.html).

**Reply by cscheid** [reply]
This is something we need to fix. I'm going to tag #4944 here so that we can track.

**Comment by jandermoreira** [comment]
For now, my workaround is a filter for `Cite` (not fully tested):

```lua
local function starts_with(text, subtext)
    return string.sub(text, 1, #subtext) == subtext
end

function Cite(cite)
    if starts_with(cite.citations[1].id, "rem") then
        cite.citations[1].prefix = pandoc.Str("Comentário")  -- remark
    elseif starts_with(cite.citations[1].id, "sol") then
        cite.citations[1].prefix = pandoc.Str("Resposta")  -- solution
    end

    return cite
end
```
